Fixed header cache for unknown values

gregw opened the pull request:
Avoid adding the unknown marker into the CACHE index. Issue introduced in #11661 fixing #11659

gregw committed:
Fixed header cache for unknown values

Avoid adding the unknown marker into the CACHE index. Issue introduced in #11661 fixing #11659

gregw committed:
Fixed header cache for unknown values

Avoid adding the unknown marker into the CACHE index. Issue introduced in #11661 fixing #11659

sbordet commented on jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java:
```diff
@@ -167,7 +167,8 @@ public class HttpParser
                 HttpField httpField = new HttpField(h, UNMATCHED_VALUE);
-                map.put(httpField.toString(), httpField);
+                map.put(h + ": ", httpField);
+                map.put(h + ":", httpField);
```
Do we really want to cache the no-space-after-column case? 
It's bad formatting by the client, so it deserves less efficiency?

gregw replied:
I added it in because I initially write a test that was just looking up "Content-Type:" (no space) and was surprised by the miss.     But I will remove for now to just fix the issue and we can think about other optimizations separately.

sbordet reviewed (changes requested):
Change look ok beside maybe not caching the no-space-after-column case, but broke other tests.

gregw committed:
Fixed header cache for unknown values

Avoid adding the unknown marker into the CACHE index. Issue introduced in #11661 fixing #11659
